### PR TITLE
COM-20083: Package submission page

### PR DESCRIPTION
**Pull request by ciastektk**
https://jira.ez.no/browse/COM-20083

add1

add2

### CODE REVIEW

File: src/AppBundle/Service/Package/PackageService.php [hunk 1]
```diff
@@ -45,16 +55,74 @@ class PackageService implements PackageServiceInterface
...
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function addPackage(array $formData): Content
+    {
+        $contentType = $this->contentTypeService->loadContentTypeByIdentifier(self::$CONTENT_TYPE_NAME);
+        $contentCreateStruct = $this->contentService->newContentCreateStruct($contentType, self::$DEFAULT_LANG_CODE);
+
+        $packageUrl = isset($formData['url']) ? $formData['url'] : '';
```
Comment (damianz5) [inline src/AppBundle/Service/Package/PackageService.php]: try `$packageUrl = $formData['url'] ?? ''`

File: src/AppBundle/Form/PackageAddType.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+     *
+     * @return array
+     */
+    private function getCategories(array $categories): array
+    {
+        $choices = [];
+
+        /**
+         * @var Tag $category
+         */
```
Comment (damianz5) [inline src/AppBundle/Form/PackageAddType.php]: use inline comment here `/** @var Tag $category */`

**Comment by ciastektk** [comment]
@SylvainGuittard could you review please? I need to add missing tests for custom validators but other parts of this PR is ready.

File: app/Resources/views/themes/ezplatform/full/package_add.html.twig [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-md-9">
+                    {{ form_start(addForm) }}
+                    <div class="form-group">
+                        {{ form_label(addForm.url) }}
+                        {{ form_widget(addForm.url) }}
+                        <div class="form-error">
+                            {{ form_errors(addForm.url) }}
```
[the remark this answers is not in the document]
Comment (ciastektk) [reply app/Resources/views/themes/ezplatform/full/package_add.html.twig]: form error

File: app/config/services/cache.yml [hunk 4]
```diff
@@ -1,6 +0,0 @@
-services:
```
[the remark this answers is not in the document]
Comment (ciastektk) [reply app/config/services/cache.yml]: I moved configuration of this service to common service config: service.yml

File: app/Resources/views/themes/ezplatform/parts/package/publication_status.html.twig [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+                Submission
+            </p>
+        </li>
+        <li class="publication__step {% if step == 'moderation' %}publication__step--active{% endif %}">
+            <p>
+                <span class="publication__step--number">2</span>
+                Moderation
+            </p>
+        </li>
+        <li class="publication__step">
```
[the remark this answers is not in the document]
Comment (ciastektk) [reply app/Resources/views/themes/ezplatform/parts/package/publication_status.html.twig]: This step is not available so you can't access to this view. This is only for layout purposes. For me the third step can stay here.

**Pull request merged**